### ISSUE TITLE
Firefox 24/25/26 no longer supported

### ISSUE THREAD

**Issue by Vangelis66** [gorhill/uBlock#2506]
Hello 
This is more of a FYI rather than a support request,  since I realise you are not expected to continue supporting old and insecure Firefox versions. 
However, on uBO's AMO pages:  

https://addons.mozilla.org/el/firefox/addon/ublock-origin/versions/
https://addons.mozilla.org/el/firefox/addon/ublock-origin/versions/beta

it is indicated that the minimum (desktop) Firefox version it would install 
(and presumably work) is "**Firefox 24.0 and later**"; the last versions 
of uBO that continue to fulfill this promise are 1.11.4 (stable channel) and 
1.11.5rc1 (beta/pre-release channel); you have recently removed from GitHub 
all 1.11.5rc XPIs but the latest (1.11.5rc6), but I had downloaded to disk 1.11.5rc2 
and can confirm that it also works with Firefox versions as old as 24. 

As a Firefox tester/enthusiast, I have kept several older versions in portable (PAF) 
formats for testing purposes; here are my test results (that prompted this report): 

Firefox versions **24.0**/**24.x.x ESR**/**25.0.1**/**26.0**

1.11.5rc1[2] _last good_
1.11.5rc3[4] _installable, non-functional_
by that I mean that no toolbar button is visible anywhere and there's no way to access 
addon's settings from within the Addons Manager (about:addons)
1.11.5rc6 _crashes Fx as soon as installed_. 
Typical crash is titled
`Crash Report [@ mozilla::layers::PLayerTransactionChild::SendPLayerConstructor ]`

Firefox version **27.0.1**

1.11.5.rc1[2] _good_
1.11.5.rc3[4] _installable, non-functional_
1.11.5rc6 _good_

I have also confirmed that 1.11.5rc6 installs and runs fine on v**28.0+**

So, to sum up, **1.11.4** & **1.11.5rc1** on AMO are the very last versions 
with true `<em:minVersion>24.0</em:minVersion>`, later RCs are unusable and 
1.11.5rc6 should really have `<em:minVersion>27.0</em:minVersion>` 

Thanks for your time reading this, again many thanks for maintaining 
my favourite Ad Blocker 👍

**Comment by camper1x** [comment]
There's no reason to keep so out-dated version in development IMO.

**Comment by gorhill** [comment]
@Vangelis66 Thanks for your investigations.

I can reproduce the crash on FF24. I found that `createWindowlessBrowser` does exist on FF24 -- even though its existence was [officially introduced with FF27](https://bugzilla.mozilla.org/show_bug.cgi?id=565388)), so the code path for `createWindowlessBrowser`-based code is taken and causes FF24 to crash.

If it had been a trivial fix I would have made it, but currently this would required to test for FF version, and it's not worth it for such an old version. So I will just increase the minimum version.

**Comment by Vangelis66** [comment]
@gorhill wrote: 

> If it had been a trivial fix I would have made it, 

... as posted already, it is only 1.11.5rc6 that crashes Fx 24/25/26, 
but uBO is already broken on those Fx versions since 1.11.5rc3; 
so, even if you had taken care of the crash by a non-trivial fix, 
you'd still have to address the earlier breakage if Fx 24/25/26 support 
was to be resumed...

As a favour, can you tell the commit 1.11.5rc2 was based on 
and, preferably, pinpoint the latest commit that would continue to function 
on Fx 24.8.1 ? (a personal interest of mine, if not asking too much... :wink: )

**Comment by gorhill** [comment]
> can you tell the commit 1.11.5rc2 was based on

https://github.com/gorhill/uBlock/commit/d7c8588c16bd9596e5412d264c9757ea732ac379

I could make rc6 work properly on FF24 ESR, when I changed [the line](https://github.com/gorhill/uBlock/blob/f6563be542cd9917ce032ef835a39f43abf1fbca/platform/firefox/bootstrap.js#L136):

    if ( appShell.createWindowlessBrowser ) {

To:

    if ( false && appShell.createWindowlessBrowser ) {

`appShell.createWindowlessBrowser` was introduced with rc3, but due to the timing issue [you reported here](https://github.com/gorhill/uBlock/issues/2502), the code path was probably not fully executing -- hence uBO left in a non-functional state. When the code path is fully taken, it crashes.

**Comment by Vangelis66** [comment]
Much obliged for your comprehensive reply... :+1: 
Take care...

**Comment by spectatorBH** [comment]
@gorhill Please for god's sake don't drop compatibility with legacy PaleMoon 26.5 (Fx 24) just yet, since you've found a trivial fix! 

If I understood Mozilla's plans correctly Fx 56 (and Fx 52 ESR) will be the last versions to support legacy extensions anyway. So until then (Q4 this year) there's no real reason to cut the oxygen for those who still depend on legacy addons and/or forced to run older Fx/PM builds.
